### DISCUSSION TITLE
Tester not reporting / reporting incorrect times.

### DISCUSSION THREAD

**Discussion by aconbere** [srozum/film_camera_tester#40] [General]
I've built one of the DIY testers and I'm having a very hard time getting the tester to work with the Sensor #2 (the simple probe). Wondering if anyone here might have some thoughts. Here is the process I'm using to run the tester.

1. I power on the unit
2. I select the simple shutter test
3. I choose 1/60th
4. I attach the probe to the back of my camera, setting the camera to 1/60
5. I hit start
6. I charge and actuate the shutter while pointing at a bright light source

I have tested the sensors and can verify that they are of the PIC0903SL type (dark is high, light is low). Therefore I'm testing with firmware 2.9.1

I observe three possible outcomes (none correct).

1. The tester reports nothing
2. the tester reports an actuation but it is WILDLY inaccurate (example: 8s for a 1/60th speed)
3. The tester goes wild reporting a series of 10 or 12us actuations over and over. 

My first thought was that the hole / sensor might be misaligned or constructed incorrectly. To test that I removed the nano and attached an oscilloscope to the sensor's A pin at the nano adapter and ground. Using the same process above I see normal square pulses (see below for an example).

![IMG_3023](https://github.com/srozum/film_camera_tester/assets/17713/ef71ef10-d1f5-4074-9f92-449236660168)

In this image we can see that the sensor used is a PIC0903SL it starts high, moves low during curtain travel, and returns high when the curtain is closed. From the scope this measures as approximately 21ms.

My next thought was that maybe something was interfering with the signal when the nano was actually plugged in, so I repeated the process with the nano attached and got the same results.

![IMG_3025](https://github.com/srozum/film_camera_tester/assets/17713/d6b5b317-1314-49b9-9d56-cc3884cbcfee)

Here are some observations from the above three cases:

1. Even when the tester reported nothing there was a clear square pulse
2. When the tester reported an inaccurate actuation the pulse was accurate
3. When the tester was going crazy there was an increase in noise (but nowhere near the floor)

At this point my best guess is that there is excess noise in my unit and it's somehow messing with the way the nano is reading that pin. Does anyone know if that pin is being read as digital or analog? Is it possible it's being used as a digital input and the little bit of noise is triggering the digital pin flipping in the arduino?

**Comment by srozum** [comment]
Hi, thanks for detailed issue description. I agree with your assumption that it looks like it reads some noise.

Make sure that your wiring is correct. Although on Sensor 2 it's called channel "A", it's a center positioned sensor and it goes to pin 3 on a socket, just like Sensor 1 channel "B".

**Comment by aconbere** [comment]
The center pin of the light sensor traces to A then Pin 3 of the Socket on the board then D3 on the arduino. I get continuity from the center pin on the sensor to D3.

**Reply by srozum** [reply]
That's it. D3 pin on Arduino is A/A (Socket A / Channel A). It should go to D4. Your cable is soldered incorrectly inside of Socket connector.

**Reply by aconbere** [reply]
Oh that's super exciting, i'll give it a go moving it here in a second.

**Reply by aconbere** [reply]
Oh wow, okay I see what you're saying. Sensor 2 calls the output A, but then attaches to the A/B pin. I definitely didn't wire that correctly!

**Reply by aconbere** [reply]
Okay we seem to be getting closer. I've now got the sensor's output going to D4 on the arduino. This seems to have mostly fixed issue 1 and 3 (it consistently reads an actuation and it no longer seems to go crazy with a ton). However it continues to read VERY long times for very short actuations (between 3 and 8 seconds). I have not tried reverting to firmware 2.9 yet.

**Reply by aconbere** [reply]
@srozum 2.9.0 did it! We have a working tester. And bonus points the shutter is accurate.

**Comment by srozum** [comment]
👍
